tutorials/environmental_data_and_visualization.md

```diff
@@ -0,0 +1,173 @@
+1658923063,85,36.80129505,-121.788972517,0.8
+1658923063,87,36.80029505,-121.789972517,0.8
+1658923063,94,36.80129505,-121.789972517,0.8
+```
+> :warning: Note that only one Environment Data file per world is allowed. Also,
+Gazebo can only load data that fits into memory.
+
+## Pre-Loading
+If you are writing SDFormat files by hand or via a script you may want to
+pre-load the data. You may use the `EnvironmentPreload` plugin for this. For
+instance we could load the first file like so (by adding the environment to the
+):
+```xml
+    <plugin
+      filename="gz-sim-environment-preload-system"
+      name="gz::sim::systems::EnvironmentPreload">
+      <data>environmental_data.csv</data>
+      <dimensions>
+        <time>timestamp</time>
+        <space>
+          <x>x</x>
+          <y>y</y>
+          <z>z</z>
+        </space>
+      </dimensions>
+    </plugin>
+```
+If you would like to use Latitude and Longitude you may set the `reference` to
+`spherical` and the `units` to `degrees`.
+```xml
+    <plugin
+      filename="gz-sim-environment-preload-system"
+      name="gz::sim::systems::EnvironmentPreload">
+      <data>environmental_data.csv</data>
+      <dimensions>
+        <time>timestamp</time>
+        <space reference="spherical" units="degrees">
+          <x>latitude</x>
+          <y>longitude</y>
+          <z>altitude</z>
+        </space>
+      </dimensions>
+    </plugin>
+```
+
+## Via GUI
+To load the data, click on the 3 dots and search for the "Environment Loader":
+@image html files/environment_data/environment_loader.png
+
+Click on it and you will see the environment loader:
+
+@image html files/environment_data/environment_loader_options.png
+
+Select the file you would want to load by clicking on the arrow next to the
+field named "Data file path".
+
+@image html files/environment_data/environment_file_loaded.png
+
+Select the appropriate fields and then click "load".
+
+@image html files/environment_data/environment_fields.png
+
+The data is now loaded into the system.
+
+# Visualizing the data
+
+To visualize the data we will load 2 plugins. You must already have loaded the
+data in. First we will search for "Environment Visualization" using the three
+dots menu (second from top):
+
+@image html files/environment_data/environment_viz_search.png
+
+Click on it. You should see some sliders.
+
+@image html files/environment_data/environment_viz_sliders.png
+
+Next search for the "Point Cloud" tool.
+
+@image html files/environment_data/environment_pc_search.png
+
+You should see a panel like so:
+
+@image html files/environment_data/environment_pc_panel.png
+
+Hit the refresh button to find the pointcloud generated b y the visuallization
+plugin. You can select the field you wish to view using the "Float vector"
+field. Once thats done, you should be able to see the data.
+
+@image html files/environment_data/environment_data_viz.png
+
+# Using custom sensors to sense the data
```

Comment:
"Custom sensor" is an overloaded phrase that, in the general Gazebo context, means custom sensor plugins. That is, the thing that was a lot of work to write in Gazebo-classic but made a lot easier in new Gazebo, which is an advertised feature of the new Gazebo. So I wouldn't use that phrase here. It would be very confusing if the user is familiar with that context.

Probably "custom environmental sensor" would be more appropriate.